File: app/src/main/java/telstra/sohan/com/sohantelstracodingtest/Presenter/Presenter.java
package telstra.sohan.com.sohantelstracodingtest.Presenter;

import android.content.Context;

import java.util.List;

import telstra.sohan.com.sohantelstracodingtest.Model.RowsDataDto;


public class Presenter implements GetDataCallBack.Presenter, GetDataCallBack.onGetDataListener {
    private GetDataCallBack.View mGetDataView;
    private Intractor mIntractor;
    public Presenter(GetDataCallBack.View mGetDataView){
        this.mGetDataView = mGetDataView;
        mIntractor = new Intractor(this);
    }
    @Override
    public void getDataFromURL(Context context, String url) {
        mIntractor.initRetrofitCall(context,url);
    }


    @Override
    public void onSuccess(String message, List<RowsDataDto> list, String title) {
        mGetDataView.onGetDataSuccess(message, list, title);
    }

    @Override
    public void onFailure(String message) {
        mGetDataView.onGetDataFailure(message);
    }
}

File: app/src/main/java/telstra/sohan/com/sohantelstracodingtest/presenter/Presenter.kt
package telstra.sohan.com.sohantelstracodingtest.presenter

import android.content.Context
import telstra.sohan.com.sohantelstracodingtest.model.FactsDataDto
import telstra.sohan.com.sohantelstracodingtest.model.Interact


class Presenter(private val mGetDataView: GetDataCallBack.View) : GetDataCallBack.Presenter, GetDataCallBack.OnGetDataListener {
    private val mInteract: Interact = Interact(this)

    override fun getCall(context: Context) {
        mInteract.initRetrofitCall(context)
    }

    override fun onSuccess(listFactsDataDto: List<FactsDataDto>, title: String) {
        mGetDataView.onGetDataSuccess(listFactsDataDto, title)
    }

    override fun onFailure(message: String) {
        mGetDataView.onGetDataFailure(message)
    }
}
